lua/lualine/components/filetype.lua

```diff
@@ -8,7 +11,25 @@ FileType.update_status = function(self)
     local ok, devicons = pcall(require, 'nvim-web-devicons')
     if ok then
       local f_name, f_extension = vim.fn.expand('%:t'), vim.fn.expand('%:e')
-      self.options.icon = devicons.get_icon(f_name, f_extension)
+      local icon, icon_highlight_group = devicons.get_icon(f_name, f_extension)
+      self.options.icon = icon
+
+      if self.options.icon and (self.options.colored or self.options.colored == nil) then
+        self.options.colored = true
+
+        local highlight_color = vim.fn.synIDattr(vim.fn.hlID(icon_highlight_group), 'fg')
```

Comment:
You can call `utils.extract_highlight_colors()` here instead of `synIDattr(hlID())`

Reply:
same idea hehe